### PR TITLE
subnet manager interface and lotus impl placeholder

### PR DESCRIPTION
**Pull request by adlrocha**
This PR introduces a `SubnetManager` to be implemented by different subnet implementations to handle the core subnet interactions. The PR also adds a placeholder for the Lotus-specific implementation so we can start dividing the work between @cryptoAtwill and I, and we can start implementing each of the operations.

### CODE REVIEW

File: src/manager/lotus_manager.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+use anyhow::Result;
+use async_trait::async_trait;
+use fvm_shared::{address::Address, econ::TokenAmount};
+use ipc_gateway::Checkpoint;
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_subnet_actor::{ConstructParams, JoinParams};
+
+use super::{SubnetInfo, SubnetManager};
+
+pub struct LotusManager {}
```
Comment (hmoniz) [inline src/manager/lotus_manager.rs:12]: Consider renaming it to `LotusSubnetManager`. Even if we ultimately want a monolithic `LotusManager`, we can construct it by the composition of the submanagers.

Comment (hmoniz) [reply src/manager/lotus_manager.rs:12]: Perhaps also rename this file just to `lotus.rs`.

File: src/manager/mod.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+///! IPC node-specific traits.
+use anyhow::Result;
+use async_trait::async_trait;
+use fvm_shared::{address::Address, econ::TokenAmount};
+use ipc_gateway::Checkpoint;
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_subnet_actor::{ConstructParams, JoinParams, Status};
+use serde::{Deserialize, Serialize};
+
+/// Trait to interact with a subnet and handle its lifecycle.
```
Comment (hmoniz) [inline src/manager/mod.rs:14]: nit: Move this stuff to a `subnet` module nested in the `manager` module.

**Review by hmoniz — APPROVED** [review]
LGTM. Just some nits regarding code structure.

**Review by cryptoAtwill — APPROVED** [review]

File: src/manager/lotus_manager.rs (hunk 1 above)
Comment (adlrocha) [reply src/manager/lotus_manager.rs:12]: Good points. Fixing it.

File: src/manager/mod.rs (hunk 2 above)
Comment (adlrocha) [reply src/manager/mod.rs:14]: You mean moving the `SubnetManager` to a `manager/subnet.rs` file?

File: src/manager/subnet.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+use serde::{Deserialize, Serialize};
+
+/// Trait to interact with a subnet and handle its lifecycle.
+#[async_trait]
+pub trait SubnetManager {
+    /// Deploys a new subnet actor on the `parent` subnet given as an input and with the
+    /// configuration passed in `ConstructParams`.
+    /// The result of the function is the ID address for the subnet actor from which the final
+    /// subet ID can be inferred.
+    fn create_subnet(parent: SubnetID, from: Address, params: ConstructParams) -> Result<Address>;
```
Comment (cryptoAtwill) [inline src/manager/subnet.rs:19]: I think the ConstructParams already contains `parent`, seems the field is not needed?

Comment (cryptoAtwill) [inline src/manager/subnet.rs:19]: This trait will need `lotus` api to interact with the node, it needs to hold `&self`, or we are initializing the lotus api every time in the call?

**Comment by adlrocha** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: MOBemDur90CBweXhoVwt.
See details on [graphite.dev](https://app.graphite.dev/github/pr/consensus-shipyard/ipc-agent/25)

**Pull request merged**